File: apps/server/registry-api/internal/entity/warehouse_connection.go
package entity

import "time"

type WarehouseConnectionStatus string

const (
	WarehouseConnectionStatusActive   WarehouseConnectionStatus = "ACTIVE"
	WarehouseConnectionStatusInactive WarehouseConnectionStatus = "INACTIVE"
	WarehouseConnectionStatusRevoked  WarehouseConnectionStatus = "REVOKED"
)

type WarehouseConnection struct {
	Id int `json:"id"`
	// The warehouse id that the connection is for
	WarehouseId string                    `json:"warehouse_id"`
	ApiKey      string                    `json:"api_key"`
	Name        string                    `json:"name"`
	Status      WarehouseConnectionStatus `json:"status"`
	CreatedAt   *time.Time                `json:"created_at"`
	UpdatedAt   *time.Time                `json:"updated_at"`
	LastUsedAt  *time.Time                `json:"last_used_at"`
}

File: packages/core/entity/warehouse-connection.ts
import { z } from "zod";

export const WarehouseConnectionStatusSchema = z.enum([
	"ACTIVE",
	"INACTIVE",
	"REVOKED",
]);

export type WarehouseConnectionStatus = z.infer<
	typeof WarehouseConnectionStatusSchema
>;

export const WarehouseConnectionSchema = z.object({
	id: z.number().int(),
	warehouse_id: z.string(),
	api_key: z.string(),
	name: z.string(),
	status: WarehouseConnectionStatusSchema,
	created_at: z.string().optional(),
	updated_at: z.string().optional(),
	last_used_at: z.string().optional(),
});

export type WarehouseConnection = z.infer<typeof WarehouseConnectionSchema>;
